presto-common/src/main/java/com/facebook/presto/common/block/DictionaryBlock.java

```diff
@@ -280,6 +280,12 @@ public long getRegionLogicalSizeInBytes(int positionOffset, int length)
         return sizeInBytes;
     }
 
+    @Override
+    public long getApproximateRegionLogicalSizeInBytes(int position, int length)
+    {
+        return (long) (dictionary.getApproximateRegionLogicalSizeInBytes(0, dictionary.getPositionCount()) * 1.0f * length / positionCount);
```

Comment:
what's the reason for `* 1.0f` ?

Reply:
You are right it doesn't matter. I removed it and it's just
```
return dictionary.getApproximateRegionLogicalSizeInBytes(0, dictionary.getPositionCount()) * length / positionCount;
```